sktime/classification/interval_based/_rise.py

```diff
@@ -374,34 +513,125 @@ def matrix_acf(x, num_cases, max_lag):
 
     """
     y = np.empty(shape=(num_cases, max_lag))
-    for lag in range(1, max_lag + 1):
+    length = x.shape[1]
+    for lag in prange(1, max_lag + 1):
         # Could just do it ourselves ... TO TEST
         #            s1=np.sum(x[:-lag])/x.shape()[0]
         #            ss1=s1*s1
         #            s2=np.sum(x[lag:])
         #            ss2=s2*s2
         #
-        y[lag - 1] = np.corrcoef(x[:, lag:], x[:, -lag])[0][1]
-        if np.isnan(y[lag - 1]) or np.isinf(y[lag - 1]):
-            y[lag - 1] = 0
+        l = length - lag
+        x1, x2 = x[:, :-lag], x[:, lag:]
+        s1 = np.sum(x1, axis=1)
+        s2 = np.sum(x2, axis=1)
+        m1 = s1 / l
+        m2 = s2 / l
+        s12 = np.sum(x1 * x2, axis=1)
+        ss1 = np.sum(x1 * x1, axis=1)
+        ss2 = np.sum(x2 * x2, axis=1)
+        v1 = ss1 - s1 * m1
+        v2 = ss2 - s2 * m2
+        v12 = s12 - s1 * m2
+        v1_is_zero, v2_is_zero = v1 <= 1e-9, v2 <= 1e-9
+        non_zero = ~v1_is_zero & ~v2_is_zero
+        # y[:, lag - 1] = np.sum((x1 - m1[:, None]) *
+        # (x2 - m2[:, None]), axis=1)
+        y[v1_is_zero & v2_is_zero, lag - 1] = 1  # Both zero variance,
+        # so must be 100% correlated
+        y[v1_is_zero ^ v2_is_zero, lag - 1] = 0  # One zero variance
+        # the other not
+        var = (v1 * v2)[non_zero]
+        y[non_zero, lag - 1] = v12[non_zero] / np.sqrt(var)
+    #     # y[lag - 1] = np.corrcoef(x[:, lag:], x[:, -lag])[0][1]
+    #     # if np.isnan(y[lag - 1]) or np.isinf(y[lag - 1]):
+    #     #     y[lag - 1] = 0
     return y
 
-
-def ps(x):
-    """Power spectrum transform.
-
-    Calculated using np function. It would be worth looking at ff implementation,
-    see difference in speed to java.
+# def ps(x):
```

Comment:
if we don't use the commented-out functions here and throughout the file, I suggest to remove them rather than comment them out, git allows you to always go back and look at the old code if necessary